optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/Joiners.java

```diff
@@ -136,6 +139,17 @@
 
     // TODO other TriJoiner methods
 
+    // ************************************************************************
+    // QuadJoiner
+    // ************************************************************************
+
+    public static <A, B, C, D, Property_> QuadJoiner<A, B, C, D> equal(
+            TriFunction<A, B, C, Property_> leftMapping, Function <D, Property_> rightMapping) {
+        return new SingleQuadJoiner<>(leftMapping, JoinerType.EQUAL, rightMapping);
+    }
```

Comment:
Yes, we need this equals(TriFunction, Function).

Open question: Do we _need_ equals(BiFunction, BiFunction) too?
I can think of use cases where this would give a performance benefit:
- Give me every A, B, C, D for which A.foo == B.foo and C.bar == D.bar and A.number * B.number * C.number * D.number == 42

But when in doubt, leave it out?